data-processing/entities/terms_v1/extractor.py

```diff
@@ -0,0 +1,123 @@
+from dataclasses import dataclass
+from typing import Iterator
+
+import pandas as pd
+import re
+import json
+
+import pysbd
+
+from common.parse_tex import DEFAULT_CONTEXT_SIZE, EntityExtractor, PlaintextExtractor
+from common.types import SerializableEntity
+
+
+###First convert the aggregate glossary json into a csv, this was done on a seperate file
+
+###Extracting all the terms from glossary csv
+glossary = pd.read_csv("gloss.csv")
+terms = glossary["terms"].tolist()
+
+###Converting terms to lower case
+terms = [item.lower() for item in terms]
+
+
+# Edit location #1: Define a Term entity; the below code is probably okay.
+@dataclass(frozen=True)
+class Term(SerializableEntity):
+    text: str
+
+
+# Edit location #2: Create a term extractor;
+# It should include your test code from last week; it just has an output
+# in a slightly different shape.
+class TermExtractor(EntityExtractor):
+    """
+    <descriptive comment>
+    """
+
+    def parse(self, tex_path: str, tex: str) -> Iterator[Term]:
+        # Extract plaintext segments from TeX
+        plaintext_extractor = PlaintextExtractor()
+        plaintext_segments = plaintext_extractor.parse(tex_path, tex)
+        ###Creating 3 columns for the dataframe, this dataframe will later be turned into final json
```

Comment:
See note above on comments.
Also, when a comment splits two chunks of code that are doing something different, separate them with an extra newline. For example:

```python
        plaintext_extractor = PlaintextExtractor()
        plaintext_segments = plaintext_extractor.parse(tex_path, tex)

        ###Creating 3 columns for the dataframe, this dataframe will later be turned into final json
        matched_terms_start = []
```